### ISSUE TITLE
Matcher for Channel.broadcast_to

### ISSUE THREAD

**Issue by TheSmartnik** [palkan/action-cable-testing#9]
Hi 😃  

Channels in ActionCable have `.broadcast_to` method ([docs](http://api.rubyonrails.org/classes/ActionCable/Channel/Broadcasting.html#method-i-broadcast_to)), which can be called both inside and outside of a channel. It also accepts object/record as a first argument. Right now there are a couple of ways to check that this method was called, but I think it'd be a good idea to add a matcher for it.

---

**Possible workarounds**

* Stub
```ruby
expect(WebNotificationsChannel).to receive (:broadcast_to)
```

* Manual construction of  a stream
```ruby
stream = WebNotificationsChannel.broadcasting_for([WebNotificationsChannel.channel_name, session])
expect { perform :action }.to have_broadcasted_to(stream)
```

**Suggestions**

* Rename current `have_broadcasted_to` matcher to `have_broadcasted`, and add above functionality to `have_broadcasted_to`.

I think that will be more straightforward. `have_broadcasted` corresponds to `#broadcast` and `have_broadcasted_to` corresponds to `.broadcast_to`. However one is an instance method, while the other one is  a class method. Maybe matchers should emphasize the difference.

* Add `channel_broadcast_to` matcher

This one clearly states the distinction between methods. However, I don't think that it is as intuitive as an option above.

In both ways, though, I don't really know what would be a good syntax for specifying channel name.

What do you think?

**Comment by palkan** [comment]
Hi!

I missed this 'cause I don't use `broadcasting_for` and prefer to construct streams by myself)

What about making `have_broadcasted_to` a little bit smarter an handle non-strings differently? I.e. to make it work like this:

```ruby
expect { perform :action }.to have_broadcasted_to(session)
# call WebNotificationsChannel.broadcasting_for([WebNotificationsChannel.channel_name, session]) under the hood
```

**Comment by TheSmartnik** [comment]
That looks good. Thanks.

And what to do in cases when `Channel.broadcast_to` is called from another channel? Should it be specified with a second argument or maybe there should be a special syntax for it?

**Comment by palkan** [comment]
> And what to do in cases when Channel.broadcast_to is called from another channel? 

Something like:

```ruby
expect { perform :action }.to have_broadcasted_to(session).from_channel(SomeChannel)
```

**Comment by palkan** [comment]
Closed by #10
